src/sinks/clickhouse.rs

```diff
@@ -17,27 +18,19 @@ use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
 use snafu::ResultExt;
 
-#[derive(Serialize, Deserialize, Debug, Copy, Clone)]
-#[serde(rename_all = "lowercase")]
-pub enum TimestampFormat {
-    Unix,
-    RFC3339,
-}
-
-#[derive(Deserialize, Serialize, Debug, Clone, Default)]
-#[serde(deny_unknown_fields)]
-pub struct EncodingConfig {
-    pub timestamp_format: Option<TimestampFormat>,
-}
-
 #[derive(Deserialize, Serialize, Debug, Clone, Default)]
 #[serde(deny_unknown_fields)]
 pub struct ClickhouseConfig {
     pub host: String,
     pub table: String,
     pub database: Option<String>,
     pub compression: Option<Compression>,
-    pub encoding: EncodingConfig,
+    #[serde(
+        deserialize_with = "EncodingConfigWithDefault::from_deserializer",
+        skip_serializing_if = "skip_serializing_if_default",
```

Comment:
Maybe I'm missing something but why do some sinks have this and others don't? My assumption is that this allows a "default" encoding?

To avoid further bikeshedding this should in theory be possible to do within a custom impl but happy to leave that out for now.

Reply:
We want to make sure a user never has to see `encoding.format` if there is only one option. So this makes sure we skip this field when it's not required in `vector generate` calls